src/loaders/OmeZarrLoader.ts

```diff
@@ -440,6 +474,9 @@ class OMEZarrLoader implements IVolumeLoader {
   }
 
   async loadVolumeData(vol: Volume, explicitLoadSpec?: LoadSpec, onChannelLoaded?: PerChannelCallback): Promise<void> {
+    // First, cancel any pending requests for this volume
+    this.requestQueue.cancelAllRequests(CHUNK_REQUEST_CANCEL_REASON);
```

Comment:
yay!

Reply:
Does this actually work?  Can you easily test this where you call loadVolumeData and it cancels previous ones without actually trying to render the stale data?